accesscontroltool-startuphook-bundle/src/main/java/biz/netcentric/cq/tools/actool/startuphook/impl/AcToolStartupHookServiceImpl.java

```diff
@@ -64,36 +81,76 @@ public void activate(BundleContext bundleContext, Config config) {
 
         boolean isCloudReady = RuntimeHelper.isCloudReadyInstance();
         Config.StartupHookActivation activationMode = config.activationMode();
-        LOG.info("AcTool Startup Hook (start level: {}  isCloudReady: {}  activationMode: {})", 
+        boolean runAsyncForMutableConent = config.runAsyncForMutableConent();
+        LOG.info("AcTool Startup Hook (start level: {}  isCloudReady: {}  activationMode: {}  runAsyncForMutableConent: {})", 
                 RuntimeHelper.getCurrentStartLevel(bundleContext),
                 isCloudReady,
-                activationMode);
+                activationMode,
+                runAsyncForMutableConent);
 
         boolean applyOnStartup = (activationMode == Config.StartupHookActivation.ALWAYS) 
                 || (isCloudReady && activationMode == Config.StartupHookActivation.CLOUD_ONLY);
 
         if(applyOnStartup) {
-            
-            try {
-
-                List<String> relevantPathsForInstallation = getRelevantPathsForInstallation();
-                LOG.info("Running AcTool with "
-                        + (relevantPathsForInstallation.isEmpty() ? "all paths" : "paths " + relevantPathsForInstallation) + "...");
-                acInstallationService.apply(null, relevantPathsForInstallation.toArray(new String[relevantPathsForInstallation.size()]),
-                        true);
-                LOG.info("AC Tool Startup Hook done. (start level " + RuntimeHelper.getCurrentStartLevel(bundleContext) + ")");
 
-                copyAcHistoryToOrFromApps(isCloudReady);
-
-            } catch (RepositoryException e) {
-                LOG.error("Exception while triggering AC Tool on startup: " + e, e);
+            if (runAsyncForMutableConent && isCompositeNodeStore) {
```

Comment:
I would allow async execution if not on composite store. The general option for activationMode is sufficient here as well.

Reply:
@kwin for the image build we cannot rely on async - `isCompositeNodeStore` is false during image build

Reply:
Right, let us a add a comment here and some output in the log to state this explicitly

Reply:
added a log message in f6f81249d0cf8142e9c87dc82cfed6e637ca5320